src/core/Pattern_match.ml

```diff
@@ -54,10 +54,9 @@ type taint_call_trace =
     }
 [@@deriving show, eq]
 
-type taint_trace = {
-  sources : (taint_call_trace * tainted_tokens) list;
-  sink : taint_call_trace;
-}
+type taint_trace =
+  (* source *)
+  (taint_call_trace * tainted_tokens * (* sink *) taint_call_trace) list
```

Comment:
Could we define a new record type, `taint_trace_item` or something?

Reply:
Why do we now have multiple sinks for a `taint_trace` ?

Reply:
Yes, we could.

Reply:
These aren't truly multiple sinks, they are multiple paths to the _same_ sink. This is because the taint going to a single sink might take multiple routes to get there, for instance if multiple different arguments are tainted.

Reply:
Thanks, got it! So we have the trace-from-source-to-current function, then the internal-trace-within-the-function (the tainted tokens), then the trace from-current-function-to-sink. Maybe this is good to explain in a comment somewhere.

Reply:
Or having a record, using some long descriptive names for the fields.